src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -338,6 +339,62 @@ public void DiagnosticSourceStartStop()
             }
         }
 
+        /// <summary>
+        /// Tests that Activity.Current flows correctly within async methods
+        /// </summary>
+        [Fact]
+        public async void ActivityCurrentFlowsWithAsyncSimple()
+        {
+            var activity = new Activity("activity").Start();
```

Comment:
Where is this activity information stored?  Do we need to Stop it / remove it somehow to prevent it from polluting other stuff?  If it's stored in AsyncLocal / ExecutionContext, then this is fine, as any changes to that will be reverted by the async method's logic.